master/buildbot/test/unit/test_changes_hgpoller.py

```diff
@@ -159,6 +159,9 @@ def test_poll_several_heads(self):
             gpo.Expect('hg', 'pull', '-b', 'default',
                        'ssh://example.com/foo/baz')
             .path('/some/dir'),
+            gpo.Expect('hg', 'pull', '-r', 'default',
```

Comment:
in order for this to be expected, you need to have the previous command to fail.
So you need a separate test that checks the usecase of older hg